### PR TITLE
MMT 3560: As a user, I can create an associated variable from a published collection

### PR DESCRIPTION
**Pull request by dmistry1**
Added functionality where a user can create a variable draft from a published collection record.
image

### CODE REVIEW

File: static/src/js/components/PublishPreview/PublishPreview.jsx [hunk 1]
```diff
@@ -243,7 +262,23 @@ const PublishPreview = () => {
             {' '}
             Record
           </Button>
-          {/* Delete Publish record button */}
+
+          {
+            derivedConceptType === 'Collection' && (
```
Comment (dmistry1) [inline static/src/js/components/PublishPreview/PublishPreview.jsx:267]: Use the consts

File: static/src/js/hooks/useIngestDraftMutation.js [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
 import { INGEST_DRAFT } from '../operations/mutations/ingestDraft'
 import getUmmVersion from '../utils/getUmmVersion'
-import errorLogger from '../utils/errorLogger'
-import useNotificationsContext from './useNotificationsContext'
 
 const useIngestDraftMutation = () => {
-  const [ingestDraftMutation] = useMutation(INGEST_DRAFT)
-
-  const navigate = useNavigate()
+  const [ingestDraft, setIngestDraft] = useState()
```
Comment (cgokey) [inline static/src/js/hooks/useIngestDraftMutation.js:7]: What is happening with these state vars?

File: static/src/js/pages/SearchPage/SearchPage.jsx [hunk 3]
```diff
@@ -134,10 +134,10 @@ const SearchPage = ({ limit }) => {
   }
 
   const buildEllipsisLinkCell = useCallback((cellData, rowData) => {
-    const { conceptId } = rowData
+    const { conceptId, revisionId } = rowData
 
     return (
-      <EllipsisLink to={`/collections/${conceptId}`}>
+      <EllipsisLink to={`/collections/${conceptId}/${revisionId}`}>
```
Comment (mandyparson) [inline static/src/js/pages/SearchPage/SearchPage.jsx:140]: says line 275 (below) is missing a test

File: static/src/js/components/PublishPreview/PublishPreview.jsx [hunk 4]
```diff
@@ -23,6 +23,8 @@ import useAppContext from '../../hooks/useAppContext'
 import useIngestDraftMutation from '../../hooks/useIngestDraftMutation'
 import removeMetadataKeys from '../../utils/removeMetadataKeys'
 import constructDownloadableFile from '../../utils/constructDownloadableFile'
+import conceptTypes from '../../constants/conceptTypes'
+import getConceptTypeByDraftConceptId from '../../utils/getConceptTypeByDraftConceptId'
```
Comment (mandyparson) [inline static/src/js/components/PublishPreview/PublishPreview.jsx:27]: This file is missing some tests

**Review by mandyparson — COMMENTED** [review]
Screenshot 2024-02-28 at 11 55 45 AM

^^Heads up on this screenshot as well after I click "save and publish" on variable draft

**Comment by dmistry1** [comment]
> Screenshot 2024-02-28 at 11 55 45 AM
> 
> ^^Heads up on this screenshot as well after I click "save and publish" on variable draft

Fixed this. good catch

File: static/src/js/components/PublishPreview/__tests__/PublishPreview.test.js [hunk 5]
```diff
@@ -531,4 +534,68 @@ describe('PublishPreview', () => {
       )
     })
   })
+
+  describe('when the collection preview is collections', () => {
+    describe('when clicking on Create Collection', () => {
+      test.only('should navigate to variable draft preview', async () => {
```
[the remark this answers is not in the document]
Comment (dmistry1) [reply static/src/js/components/PublishPreview/__tests__/PublishPreview.test.js]: Fixed

File: static/src/js/pages/SearchPage/SearchPage.jsx (hunk 3 above)
Comment (dmistry1) [reply static/src/js/pages/SearchPage/SearchPage.jsx:140]: This was fixed by removing the `.only` from this test file

Comment (dmistry1) [reply static/src/js/pages/SearchPage/SearchPage.jsx:140]: line 275 is a sortFn, there is an test for that. I have verified that making this revisionId change did not cause that line to be uncovered. 

```
{
  dataKey: 'longName',
  title: 'Long Name',
  className: 'col-auto',
  dataAccessorFn: buildEllipsisTextCell,
  sortFn: (_, order) => sortFn('longName', order)
},
```

**Review by cgokey — APPROVED** [review]

**Review by tle1989 — APPROVED** [review]

File: static/src/js/components/PublishPreview/__tests__/__mocks__/publishPreview.js [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
... Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text  Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Text Mock Very Long Text Mock Very Long Text Mock Very LongText Mock Very Long Text Mock Very Long Text Mock Very Long Text  Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text  Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text  Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text  Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Long Text Mock Very Text Mock Very Long Text Mock Very Long Text Mock Very LongText Mock Very Long',
```
Comment (trevorlang) [inline static/src/js/components/PublishPreview/__tests__/__mocks__/publishPreview.js:11]: Maybe use some Lorem Ipsum or a real description here 😆

**Review by trevorlang — APPROVED** [review]
Approved with some minor feedback

File: static/src/js/components/PublishPreview/__tests__/__mocks__/publishPreview.js (hunk 6 above)
Comment (dmistry1) [reply static/src/js/components/PublishPreview/__tests__/__mocks__/publishPreview.js:11]: Ooop... This is what I get get for being lazy and getting a test record from SIT, SMH. Fixed.

**Pull request merged**